### PR TITLE
Recent Imports

### PR DESCRIPTION
**Pull request by shivam-tripathi**
Recent Imports

### CODE REVIEW

File: src/func/imports/recent-imports.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+						.offset(offset)
+						.as('link'),
+					'link.import_id',
+					'bookbrainz.import.id'
+				);
+
+	/* Contruct importHolder object (holds importIds classified by their types)
+		object of form {type: []} */
+	const importHolder = _.values(entityTypes).reduce(
+		(holder: Object, type: string) => _.assign(holder, {[type]: []}), {}
```
Comment (LordSputnik) [inline src/func/imports/recent-imports.js]: No need to use lodash here now that we have ES6:

    const importHolder = entityTypes.values().reduce(
    	(holder: Object, type: string) => ({...holder, {[type]: []}), {}
    );

File: src/func/imports/recent-imports.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+export default async function getRecentImports(
+	orm: Object, transacting: Transaction, limit: number = 10,
+	offset: number = 0
+) {
+	// Fetch most recent ImportIds classified by importTypes
+	const {importHolder: recentImportIdsByType, timeStampMap} =
+		await getRecentImportIdsByType(transacting, limit, offset);
+
+	const importTypes: Array<EntityTypeString> = _.values(entityTypes);
```
Comment (LordSputnik) [inline src/func/imports/recent-imports.js]: Use native ES6 here too - `entityTypes.values()`

File: src/func/imports/recent-imports.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+				getRecentImportsByType(transacting, type,
+					recentImportIdsByType[type]))
+		)
+	);
+
+	const defaultAliasIds = _.map(recentImports, 'default_alias_id');
+	const defaultAliases =
+		await getAliasByIds(transacting, defaultAliasIds);
+	// Append defaultAlias objects to the recentImports
+	defaultAliases.forEach((defaultAlias, index) => {
```
Comment (LordSputnik) [inline src/func/imports/recent-imports.js]: Can probably merge this into the loop below, if you return a map of aliasId, alias pairs, then just assign `recentImport.defaultAlias = defaultAliases[recentImport.defaultAliasId]`.

File: src/func/imports/recent-imports.js (hunk 1 above)
Comment (shivam-tripathi) [reply src/func/imports/recent-imports.js]: Oh nice! 😃

File: src/func/imports/recent-imports.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+ * @returns {Promise<Object<Array>>} - Returns an object containing arrays of
+ * 		importIds of various importTypes
+ */
+async function getRecentImportIdsByType(
+	transacting: Object,
+	limit: number,
+	offset: number
+) {
+	// Extract recent imports, types and import timeStamp
+	const recentImportIdAndType: Object =
```
[the remark this answers is not in the document]
Comment (shivam-tripathi) [reply src/func/imports/recent-imports.js]: My belief was that when import data exceeds a limit, joining two big tables and then limiting them would be an expensive process compared to first running a query to fetch smaller chunk and then running join. Am I wrong in this regard?

File: src/func/imports/recent-imports.js (hunk 3 above)
Comment (shivam-tripathi) [reply src/func/imports/recent-imports.js]: Right, should have paid more attention.

File: src/func/imports/recent-imports.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (shivam-tripathi) [reply src/func/imports/recent-imports.js]: Also, in this query I think I've used alias (see the subquery)

File: src/func/imports/recent-imports.js (hunk 1 above)
Comment (shivam-tripathi) [reply src/func/imports/recent-imports.js]: Doesn't work somehow in the present setup, probably need to update the babel version for transpiling the es6 version or something

File: src/func/imports/recent-imports.js (hunk 2 above)
Comment (shivam-tripathi) [reply src/func/imports/recent-imports.js]: Same

**Comment by LordSputnik** [comment]
Superseded by #129

**Pull request closed without merging**